rest-openehr/src/main/java/org/ehrbase/rest/util/AuthHelper.java

```diff
@@ -47,34 +47,29 @@ private AuthHelper() {}
      */
     public static String getCurrentAuthenticatedUsername(HttpServletRequest request) {
         Principal principal = request.getUserPrincipal();
-        String username;
-
-        // Check if the principal is null and get the principal from the SecurityContext if necessary
-        if (principal == null) {
-            principal = SecurityContextHolder.getContext().getAuthentication();
-        }
-
-        username = Optional.ofNullable(principal)
+        return Optional.ofNullable(principal)
                 .filter(AbstractAuthenticationToken.class::isInstance)
                 .map(AbstractAuthenticationToken.class::cast)
                 .map(AbstractAuthenticationToken::getPrincipal)
                 .filter(DecodedJWT.class::isInstance)
                 .map(DecodedJWT.class::cast)
                 .map(DecodedJWT::getSubject)
-                .orElse(EMPTY);
-
-        if (isBlank(username)
-                && request.getHeader(AUTHORIZATION) != null
-                && request.getHeader(AUTHORIZATION).startsWith("Basic")) {
-            username = getBasicAuthUsername(request);
-        }
-
-        // If the username is still blank, use the name from the Principal
-        if (isBlank(username) && principal != null) {
-            username = principal.getName();
-        }
+                .filter(StringUtils::isNotBlank)
+                .orElseGet(() -> {
+                    Function<Principal, String> username = p -> p != null ? p.getName() : null;
+                    if (hasBasicAuthHeader(request)) {
+                        return Optional.ofNullable(getBasicAuthUsername(request))
```

Comment:
`username.apply(principal)` in   `.orElse(username.apply(principal))`
is performed unconditionally,
so you could get rid of the Function and retrieve it beforehand, 
or do it via `::orElseGet`.
`AuthHelper::hasBasicAuthHeader` could be dropped, because it is included in `AuthHelper::getBasicAuthUsername`:

```
return Optional.of(request)
.map(AuthHelper::getBasicAuthUsername)
.filter(StringUtils::isNotBlank)
.orElseGet(()-> Optional.ofNullable(principal).map(Principal::getName).orElse(null));
```

Reply:
done